pytradfri/group.py

```diff
@@ -24,85 +32,116 @@
     ROOT_GROUPS,
 )
 from .error import ColorError
-from .resource import ApiResource
+from .resource import ApiResource, ApiResourceResponse, TypeRaw
+
+if TYPE_CHECKING:
+    from .gateway import Gateway
+
+
+class GroupResponse(ApiResourceResponse):
+    """Represent API response for a blind."""
+
+    color_hex: Optional[str] = Field(alias=ATTR_LIGHT_COLOR_HEX)
+    dimmer: int = Field(alias=ATTR_LIGHT_DIMMER)
+    group_members: Optional[dict[str, Any]] = Field(alias=ATTR_GROUP_MEMBERS)
+    id: int = Field(alias=ATTR_ID)
+    mood_id: str = Field(alias=ATTR_MOOD)
+    state: int = Field(alias=ATTR_DEVICE_STATE)
 
 
 class Group(ApiResource):
     """Represent a group."""
 
-    def __init__(self, gateway, raw):
+    _model_class: type[GroupResponse] = GroupResponse
+    raw: GroupResponse
+
+    def __init__(self, gateway: Gateway, raw: TypeRaw):
         """Create object of class."""
         super().__init__(raw)
+        self.raw = GroupResponse(**raw)
         self._gateway = gateway
 
     @property
-    def path(self):
+    def path(self) -> list[str]:
         """Path."""
-        return [ROOT_GROUPS, self.id]
+        return [ROOT_GROUPS, str(self.id)]
 
     @property
-    def state(self):
+    def state(self) -> bool:
         """Boolean representing the light state of the group."""
-        return self.raw.get(ATTR_DEVICE_STATE) == 1
+        return self.raw.state == 1
 
     @property
-    def dimmer(self):
+    def dimmer(self) -> int:
         """Dimmer value of the group."""
-        return self.raw.get(ATTR_LIGHT_DIMMER)
+        return self.raw.dimmer
 
     @property
-    def hex_color(self):
+    def hex_color(self) -> str | None:
         """Return hex color."""
-        return self.raw.get(ATTR_LIGHT_COLOR_HEX)
+        if self.raw.color_hex:
+            return self.raw.color_hex
+
+        return None
 
     @property
-    def member_ids(self):
-        """Members of this group."""
-        info = self.raw.get(ATTR_GROUP_MEMBERS, {})
+    def member_ids(self) -> list | list[int]:
```

Comment:
We don't need a special type case for returning an empty list.
```suggestion
    def member_ids(self) -> list[int]:
```